app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -197,10 +244,74 @@ open class ProfilesViewModel(
           _selectedProfile.update { currentProfile ->
             currentProfile?.copy(profilePictureUrl = null)
           }
+          updateProfile(selectedProfile.value!!)
         },
         onFailure = { e -> handleError(e) })
   }
 
+  /**
+   * Converts an image URI to a processed Bitmap, cropping the image to a square at the center.
+   *
+   * @param context The context used to access the content resolver.
+   * @param imageUri The URI of the image to be converted.
+   * @param maxResolution The maximum resolution for the output Bitmap. Default is 600.
+   * @return A Bitmap representing the processed image, or null if an error occurs.
+   */
+  private fun imageUriToBitmap(context: Context, imageUri: Uri, maxResolution: Int = 600): Bitmap? {
+    return try {
+      // Open an InputStream from the URI
+      val inputStream: InputStream? = context.contentResolver.openInputStream(imageUri)
```

Comment:
Considering further implementation of a cropping library, this will have to be modified : you don't want to strictly crop a 600x600 frame out of the original image, but rather let the user choose the part he wants, and then degrade the quality of its selection to obtain a 600x600 version of what he selected.